dipy/tracking/life.py

```diff
@@ -306,9 +307,7 @@ def voxel2streamline(streamline, transformed=False, affine=None,
 
     if unique_idx is None:
         all_coords = np.concatenate(transformed_streamline)
-        unique_idx = unique_rows(all_coords.astype(int))
-    else:
-        unique_idx = unique_idx
+        unique_idx = unique_rows(np.round(all_coords).astype(np.intp))
```

Comment:
How about leaving the cast to the later line, to avoid users getting errors from passed `unique_idx` of wrong data type.  As in:

```
        unique_idx = unique_rows(np.round(all_coords))

return _voxel2streamline(transformed_streamline, unique_idx.astype(np.intp))
```